.github/workflows/build-and-test.yaml

```
@@ -0,0 +1,161 @@
+    -
+      name: Import Luxembourg
+      run : |
+        docker  volume  create  ${VOLUME}
+        docker  run  --rm  --shm-size=128M  -v ${VOLUME}:${MOUNT}  -e UPDATES=enabled  ${IMAGE}:${TAG}  import
+    -
+      name: Start server
+      run : |
+        docker  run  --rm  --shm-size=128M  -v ${VOLUME}:${MOUNT}  -e UPDATES=enabled  -p 80:80  -d  --name ${CONTAINER}  ${IMAGE}:${TAG}  run
+        sleep 30
+    -
+      name: Download tiles
+      run : |
+        curl  http://localhost/tile/0/0/0.png            --fail  -o 000.png
+        curl  http://localhost/tile/1/0/0.png            --fail  -o 100.png
+        curl  http://localhost/tile/1/0/1.png            --fail  -o 101.png
+        curl  http://localhost/tile/1/1/0.png            --fail  -o 110.png
+        curl  http://localhost/tile/1/1/1.png            --fail  -o 111.png
+        curl  http://localhost/tile/18/138474/85459.png  --fail  -o empty.png
+        curl  http://localhost/tile/18/135536/89345.png  --fail  -o example.png
+    -
+      name: Upload tiles
+      uses: actions/upload-artifact@v3
+      with:
+        name: tiles
+        path: '*.png'
+    -
+      name: Verify tiles
+      run : |
+        sha1sum  *.png
+        sha1sum  --check  <<EOF
+        c226ca747874fb1307eef853feaf9d8db28cef2b *empty.png
+        EOF
+        tiles=(`ls *.png`)
+        for ((i=0; i<${#tiles[@]}; i++)) ; do
+          if [ `file  --brief  --mime-type  "${tiles[$i]}"` != 'image/png' ] ; then
+            >&2  echo  "ERROR: ${tiles[$i]} is not a image/png file"
+            exit 1
+          fi
+          for ((j=i+1; j<${#tiles[@]}; j++)) ; do
+            if ( diff  "${tiles[$i]}"  "${tiles[$j]}" ) ; then
+              >&2  echo  "ERROR: ${tiles[$i]} is identical to ${tiles[$j]}"
+              exit 2
+            fi
+          done
+        done
+    -
+      name: Cleanup
+      run : |
+        docker  rm  --force  --volumes  ${CONTAINER}
+        docker  volume  rm  --force  ${VOLUME}
+        docker  rmi  --force  ${IMAGE}:${TAG}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs:
+    - test
+    if: ${{ github.event_name != 'pull_request' }}
+    steps:
+    -
+      name: Checkout
+      uses: actions/checkout@v3
+    -
+      name: Environment
+      run : |
+        echo  IMAGE=$(echo ${{ env.IMAGE }} | tr '[:upper:]' '[:lower:]')  >>$GITHUB_ENV
+        echo  DOCKERHUB_IMAGE=$([ "${{ secrets.DOCKERHUB_USERNAME }}" != '' ] && [ "${{ secrets.DOCKERHUB_PASSWORD }}" != "" ] && echo "$IMAGE")  >>$GITHUB_ENV
+    -
+      name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          ${{ env.DOCKERHUB_IMAGE }}
+          ghcr.io/${{ env.IMAGE }}
+        tags: |
+          type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
```

do we want to lead with a `v` as we currently do in [docker hub](https://hub.docker.com/r/overv/openstreetmap-tile-server/tags)? 

```suggestion
          type=semver,pattern=v{{version}}
          type=semver,pattern=v{{major}}.{{minor}}
          type=semver,pattern=v{{major}}
```

---

We could do that possibly. I personally do prefer it without, but that's my subjective opinion of course.

I think with docker images it is more common that version tags are without a leading `v`. Examples without `v`:
- [alpine](https://hub.docker.com/_/alpine?tab=tags)
- [nginx](https://hub.docker.com/_/nginx?tab=tags)
- [php](https://hub.docker.com/_/php?tab=tags)

The [example configuration](https://github.com/docker/metadata-action/blob/f2a13332ac1ce8c0a71aeac48a150dbb1838ab67/README.md?plain=1#L312-L314) of the `docker/metadata-action` action that I use has it without.

---

I am only worried about the changing the strategy in docker hub. But as we are not doing that with this pr i am fine with the change